core/commands/swarm.go

```diff
@@ -128,7 +129,11 @@ var swarmPeersCmd = &cmds.Command{
 
 			buf := new(bytes.Buffer)
 			for _, info := range ci.Peers {
-				fmt.Fprintf(buf, "%s/ipfs/%s", info.Addr, info.Peer)
+				if strings.Contains(info.Addr, "/p2p-circuit/") {
+					fmt.Fprintf(buf, "%s", info.Addr)
+				} else {
+					fmt.Fprintf(buf, "%s/ipfs/%s", info.Addr, info.Peer)
```

Comment:
Basically `/ipfs/QmId` serves two purposes:

1. Routing information: An alias for an internet address (stored in the DHT): `/ip4/xxx.xxx.xxx.xxx/tcp/1234`
2. Identity information: The key of the target machine.

In the `RemoteMultiaddr` case, the second `ipfs` address is routing information.

We'd actually run into the same problem something returned `/ipfs/QmId` as the entire `RemoteMultiaddr` (which would actually make sense if you had a lower-level VPN that understood IPFS addresses).

---

Actually, that brings up a good point. Maybe we should just be checking to see if the last address is an `/ipfs/` address and, if it is, assert that the `QmIds` match instead of appending something. Thoughts @vyzo?